cloud/cloud_env_impl.h

```diff
@@ -366,6 +407,9 @@ class CloudEnvImpl : public CloudEnv {
   std::mutex files_to_delete_mutex_;
   std::chrono::seconds file_deletion_delay_ = std::chrono::hours(1);
   std::unordered_map<std::string, int> files_to_delete_;
+
+  // mutex to protect updates against cloud_env_options
+  port::RWMutex cloud_env_option_mutex_;
```

Comment:
Introducing this mutex means we need to lock `cloud_env_option_mutex_` in read mode for every access to `cloud_env_options`, which is definitely not the case today.

I don't think we need to mutate cookie ever, though. See my other comments.

Reply:
Makes sense. In that case, we need to rename `cookie` to `cookie_when_open` as well, since it might point to switched CloudManifestFile.